File: py_code/435_Non-overlapping Intervals.py
from typing import List

class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
      intervals.sort(key=lambda x:x[1])
      temp=intervals[0][1]
      count=0
      for i in range(1,len(intervals)):
        if intervals[i][0]<temp:
          count+=1
        else:
          temp=intervals[i][1]
        pass
      print(count)
      return count


s=Solution()
s.eraseOverlapIntervals([[1,100],[11,22],[1,11],[2,12]])

File: js_code/435. Non-overlapping Intervals.js
/**
 * @param {number[][]} intervals
 * @return {number}
 */
 var eraseOverlapIntervals = function(intervals) {
    intervals =intervals.sort((a,b)=>a[1]-b[1]);
    let temp=intervals[0][1],counts=0;
    for(let i=1; i<intervals.length; i++) {
      if(intervals[i][0]<temp) {
        counts+=1;
      }
      else{
        temp=intervals[i][1];
      }
    }
    console.log(intervals,temp,counts);
    return counts;
};


eraseOverlapIntervals([[1,100],[11,22],[1,11],[2,12]])